src/services/test-helpers/mock/mockApi.ts

```diff
@@ -517,6 +519,159 @@ const auctionsWinningsAt = () =>
+			owner: accountIdOne,
+			issue: accountIdTwo,
+			admin: accountIdTwo,
+			freezer: accountIdTwo,
+			supply: assetTBalanceOne,
+			deposit: balanceOfTwo,
+			minBalance: rococoRegistry.createType('u64', 10000),
+			isSufficient: trueBool,
+			accounts: rococoRegistry.createType('u32', 10),
+			sufficients: rococoRegistry.createType('u32', 15),
+			approvals: rococoRegistry.createType('u32', 20),
+			isFrozen: falseBool,
+		};
+
+		return rococoRegistry.createType('AssetDetails', responseObj);
+	});
+
+const assetsMetadata = () =>
+	Promise.resolve().then(() => {
+		const responseObj = {
+			deposit: balanceOfTwo,
+			name: rococoRegistry.createType('Bytes', 'statemint'),
+			symbol: rococoRegistry.createType('Bytes', 'DOT'),
+			decimals: rococoRegistry.createType('u8', 10),
+			isFrozen: falseBool,
+		};
+
+		return rococoRegistry.createType('AssetMetadata', responseObj);
+	});
+
+const assetBalanceObjOne = {
+	balance: assetTBalanceOne,
+	isFrozen: falseBool,
+	isSufficient: trueBool,
+};
+
+const assetBalanceObjTwo = {
+	balance: assetTBalanceTwo,
+	isFrozen: trueBool,
+	isSufficient: trueBool,
+};
+
+const assetBalanceObjThree = {
+	balance: assetTBalanceTwo,
+	isFrozen: falseBool,
+	isSufficient: falseBool,
+};
+
+const assetBalanceFactory = {
+	'10': rococoRegistry.createType('AssetBalance', assetBalanceObjOne),
+	'20': rococoRegistry.createType('AssetBalance', assetBalanceObjTwo),
+	'30': rococoRegistry.createType('AssetBalance', assetBalanceObjThree),
+};
+
+const assetAccountStorageKeyOne = statemintTypeFactory.storageKey(
+	10,
+	'AssetId',
+	statemintApiV1.query.assets.asset,
+	accountIdOne
+);
+
+const assetAccountStorageKeyTwo = statemintTypeFactory.storageKey(
+	20,
+	'AssetId',
+	statemintApiV1.query.assets.asset,
+	accountIdOne
+);
+
+const assetAccountStorageKeyThree = statemintTypeFactory.storageKey(
+	30,
+	'AssetId',
+	statemintApiV1.query.assets.account,
+	accountIdOne
+);
+
+const assetsAccountKeysAt = () =>
+	Promise.resolve().then(() => {
+		return [
+			assetAccountStorageKeyOne,
+			assetAccountStorageKeyTwo,
+			assetAccountStorageKeyThree,
+			assetAccountStorageKeyThree,
+			assetAccountStorageKeyThree,
+		];
+	});
+
+/**
+ * @param assetId options are 10, 20, 30
+ */
+interface AssetsAccount {
+	// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+	[key: string]: any;
+}
+
+const assetsAccount: AssetsAccount = (
+	assetId: number | AssetId,
+	_address: string
+) => {
+	const id = typeof assetId === 'number' ? assetId : assetId.toNumber();
+
+	return Promise.resolve().then(() => {
+		switch (id) {
+			case 10:
+				return assetBalanceFactory[10];
+			case 20:
+				return assetBalanceFactory[20];
+			case 30:
+				return assetBalanceFactory[30];
+			default:
+				return;
+		}
+	});
+};
+
+assetsAccount.extend = function (args: AssetsAccount) {
+	for (const i in args) {
+		// eslint-disable-next-line  @typescript-eslint/no-unsafe-assignment
+		this[i] = args[i];
+	}
+	return this;
```

Comment:
```suggestion
        return Object.assign(this, args);
```

Though that alone looks like code smell, would just use `Object.assign` over `x.extend` and drop the method entirely.

Reply:
Fixed per comment below.